File: SourceCode/WebSite/API Services/BusinessObjects/SiteExceptionInfo.vb
Imports TotalRecall.Enumerations

Public Class SiteExceptionInfo
    Inherits WebBusinessInfo

    Public Property Type() As String
    Public Property ErrorType() As ErrorTypes
    Public Property ErrorMessage() As String
    Public Property ErrorDetails() As String
    Public Property ValidationErrors() As List(Of ValidationError)
    Public Property StatusCode() As Integer

    Public Sub New()

        Me.ErrorType = ErrorTypes.General
        Me.Type = "SiteException"

    End Sub

    Public Shared Function Fill(errorMessage As String, errorType As ErrorTypes) As SiteExceptionInfo

        Dim x As New SiteExceptionInfo()

        x.ErrorMessage = errorMessage
        x.ErrorType = errorType

        Return x

    End Function

End Class

File: SourceCode/WebSite/BusinessObjects/SiteExceptionInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace TotalRecall.BusinessObjects
{
    public class SiteExceptionInfo : WebBusinessInfo
    {
        public string Type { get; set; }

        public ErrorTypes ErrorType { get; set; }

        public string ErrorMessage { get; set; }

        public string ErrorDetails { get; set; }

        public List<ValidationError> ValidationErrors { get; set; }

        public int StatusCode { get; set; }

        public SiteExceptionInfo()
        {
            this.ErrorType = ErrorTypes.General;
            this.Type = "SiteException";
        }

        public static SiteExceptionInfo Fill(string errorMessage, ErrorTypes errorType)
        {
            SiteExceptionInfo x = new SiteExceptionInfo();
            x.ErrorMessage = errorMessage;
            x.ErrorType = errorType;

            return x;
        }

    }

    public class ValidationError : WebBusinessInfo
    {
        public string PropertyName { get; set; }

        public string Error { get; set; }
    }

    public enum ErrorTypes : byte
    {
        Error = 0,
        General = 1,
        Fatal = 2,
        Validation = 3,
        Warning = 4
    }
}
